Dockerfile

```diff
@@ -237,13 +237,13 @@ RUN MAX_JOBS=2  pip --verbose wheel --no-deps flash-attn==${FLASH_ATT_VERSION} \
 
 ## Install auto-gptq ###########################################################
 FROM python-builder as auto-gptq-installer
-ARG AUTO_GPTQ_REF=ccb6386ebfde63c17c45807d38779a93cd25846f
+ARG AUTO_GPTQ_REF=896d8204bc89a7cfbda42bf3314e13cf4ce20b02
 
 WORKDIR /usr/src/auto-gptq-wheel
 
 # numpy is required to run auto-gptq's setup.py
 RUN pip install numpy
-RUN DISABLE_QIGEN=1 pip wheel git+https://github.com/AutoGPTQ/AutoGPTQ@${AUTO_GPTQ_REF} --no-cache-dir --no-deps --verbose
+RUN BUILD_CUDA_EXT=1 COMPILE_MARLIN=1 DISABLE_QIGEN=1 pip wheel git+https://github.com/AutoGPTQ/AutoGPTQ@${AUTO_GPTQ_REF} --no-cache-dir --no-deps --verbose
```

Comment:
@cyang49 do you know if we would get the Marlin kernel if we pip install from the auto-gptq wheel (e.g. latest version 0.7.1)?

Reply:
I'm not sure. I did a quick test in a local python env and got an error while trying to import marlin
```>>> import torch
>>> import autogptq_marlin_cuda
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: /net/storage149/autofs/css22/ccyang/miniconda3-netsres/envs/vllm/lib/python3.11/site-packages/autogptq_marlin_cuda.cpython-311-x86_64-linux-gnu.so: undefined symbol: _ZN2at4_ops19empty_memory_format4callEN3c108ArrayRefINS2_6SymIntEEESt8optionalINS2_10ScalarTypeEES6_INS2_6LayoutEES6_INS2_6DeviceEES6_IbES6_INS2_12MemoryFormatEE
```
I can rebuild the image locally and see if it works in the TGIS container

Reply:
According to AutoGPTQ readme, the package depends on torch 2.2.1+cu121.
I see that TGIS uses 2.2.0. Will that be a problem..?

Reply:
We can update torch too. May need to also update flash attention version but that's fine

Reply:
I tried rebuilding the image with prebuilt auto-gptq 0.7.1 but it doesn't work
```
Shard 0:
Shard 0:   File "/opt/tgis/lib/python3.11/site-packages/text_generation_server/utils/layers.py", line 192, in get_linear
Shard 0:     linear = (QuantLinear if not use_gptq_cuda else GPTQ_CUDA_LINEAR)(
Shard 0:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Shard 0:
Shard 0:   File "/opt/tgis/lib/python3.11/site-packages/text_generation_server/utils/gptq/marlin.py", line 145, in __init__
Shard 0:     self.B = autogptq_marlin_cuda.gptq_repack(qweight)
Shard 0:              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
Shard 0:
Shard 0: AttributeError: 'function' object has no attribute 'gptq_repack'
```

Reply:
I updated the Dockerfile to pull v0.7.1 for the build-from-source